File: src/main/java/plus/maa/backend/controller/ArkLevelController.java
package plus.maa.backend.controller;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import plus.maa.backend.common.annotation.AccessLimit;
import plus.maa.backend.controller.response.ArkLevelInfo;
import plus.maa.backend.controller.response.MaaResult;
import plus.maa.backend.service.ArkLevelService;

import java.util.List;

/**
 * @author john180
 */
@Slf4j
@RestController
@RequiredArgsConstructor
public class ArkLevelController {
    private final ArkLevelService arkLevelService;

    @GetMapping("/arknights/level")
    public MaaResult<List<ArkLevelInfo>> getLevels() {
        return MaaResult.success(arkLevelService.getArkLevelInfos());
    }

    @GetMapping("/test")
    @AccessLimit(times = 2,second=100)
    public MaaResult getLevels111() {
        return MaaResult.success("122321",null);
    }
}

File: src/main/java/plus/maa/backend/controller/ArkLevelController.kt
package plus.maa.backend.controller;

import java.util.List;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import plus.maa.backend.controller.response.copilot.ArkLevelInfo;
import plus.maa.backend.controller.response.MaaResult;
import plus.maa.backend.service.ArkLevelService;

/**
 * @author john180
 */
@RestController
@RequiredArgsConstructor
@Tag(name = "ArkLevelController", description = "关卡数据管理接口")
public class ArkLevelController {
    private final ArkLevelService arkLevelService;

    @Operation(summary = "获取关卡数据")
    @ApiResponse(description = "关卡数据")
    @GetMapping("/arknights/level")
    public MaaResult<List<ArkLevelInfo>> getLevels() {
        return MaaResult.success(arkLevelService.getArkLevelInfos());
    }

}
